build.c

```diff
@@ -201,46 +209,52 @@ buildadd(struct node *n)
 	e->flags &= ~FLAG_CYCLE;
 }
 
-static void
+/* returns length of printed status */
+static int
 printstatus(void)
 {
 	const char *fmt;
 	struct timespec endtime;
+	int len = 0;
```

Comment:
Even though printf returns `int`, I think it would be better to store the total length in `size_t` here.

I'm also wondering about the case where printf fails and returns -1 (this could happen if, for example, samurai output was redirected to a file on a disk that runs out of space).

Reply:
I'll think about how to deal with the -1 case, but in terms of storing it as a size_t, the parameter for * in a printf format string is supposed to be an int anyway so I'd end up having to cast len to int anyway, wouldn't that remove any benefit of storing it in a size_t?

Reply:
Ah, yes, that's a good point.

Reply:
I added a new variable to assign the return value of printf to, and if it's above 0 I add it to len, that should remove possible problems from printf returning -1.